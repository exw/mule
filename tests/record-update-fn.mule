fn x. x where { y = {} }
