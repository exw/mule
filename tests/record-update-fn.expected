
Desugared:

	fn x. _.y:=_ x {}

inferred type:

	{, ...a} -> {y : {}, ...a}

Runtime term:

	fn. (fn. fn. v1 where { y = v0 }) v0 {}

Evaluated:

	fn. (fn. fn. v1 where { y = v0 }) v0 {}
